File: gestaolegal/models/lembrete.py
from dataclasses import dataclass
from datetime import datetime
from typing import TYPE_CHECKING

if TYPE_CHECKING:
    from gestaolegal.models.user import UserInfo


@dataclass
class Lembrete:
    id_do_criador: int
    id_caso: int
    id_usuario: int
    data_criacao: datetime
    data_lembrete: datetime
    descricao: str
    status: bool
    num_lembrete: int | None = None
    id: int | None = None


@dataclass
class LembreteListItem:
    id: int
    num_lembrete: int | None
    id_caso: int
    data_criacao: datetime
    data_lembrete: datetime
    descricao: str
    status: bool
    criador: "UserInfo | None"
    usuario: "UserInfo | None"

File: web/src/lib/types/lembrete.ts
export type Lembrete = {
	id: number;
	num_lembrete: number | null;
	id_caso: number;
	data_criacao: string;
	data_lembrete: string;
	descricao: string;
	status: boolean;
	criador?: { id: number; nome: string } | null;
	usuario?: { id: number; nome: string } | null;
};
